src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,300 @@
+            feature = audio[start_sample:end_sample]
+            batch.append(feature)
+        batch = pad_sequence(batch, batch_first=True, padding_value=padding_value)
+
+        return batch, ext_beatstep
+
+    def single_preprocess(
+        self,
+        beatstep,
+        feature_tokens=None,
+        audio=None,
+        n_bars=None,
+    ):
+        """Preprocessing method for a single sequence."""
+
+        if feature_tokens is None and audio is None:
+            raise ValueError("Both `feature_tokens` and `audio` can't be None at the same time!")
+        if feature_tokens is not None:
+            if len(feature_tokens.shape) != 1:
+                raise ValueError(f"Expected `feature_tokens` shape to be (n, ) but found {feature_tokens.shape}")
+        if audio is not None:
+            if len(audio.shape) != 1:
+                raise ValueError(f"Expected `audio` shape to be (n, ) but found {feature_tokens.shape}")
+        n_bars = self.n_bars if n_bars is None else n_bars
+
+        if beatstep[0] > 0.01:
+            warnings.warn(f"Inference Warning : beatstep[0] is not 0 ({beatstep[0]}). all beatstep will be shifted.")
+            beatstep = beatstep - beatstep[0]
+
+        if self.use_mel:
+            batch, ext_beatstep = self.preprocess_mel(
+                audio,
+                beatstep,
+                n_bars=n_bars,
+                padding_value=self.padding_value,
+            )
+        else:
+            raise NotImplementedError("use_mel must be True")
+
+        return batch, ext_beatstep
+
+    def __call__(
+        self,
+        raw_audio: Union[np.ndarray, List[float], List[np.ndarray]],
+        audio_sr: int,
+        steps_per_beat: int = 2,
+        return_tensors: Optional[Union[str, TensorType]] = "pt",
+        **kwargs,
+    ) -> BatchFeature:
+        """
+        Args:
+        Main method to featurize and prepare for the model one sequence. Please note that `Pop2PianoFeatureExtractor`
+        only accepts one raw_audio at a time.
+            raw_audio (`np.ndarray`, `List`):
+                Denotes the raw_audio.
+            audio_sr (`int`):
+                Denotes the Sampling Rate of `raw_audio`.
+            steps_per_beat (`int`, *optional*, defaults to 2):
+                Denotes Steps per beat.
+            return_tensors (`str` or [`~utils.TensorType`], *optional*, defaults to 'pt'):
+                If set, will return tensors instead of list of python integers. Acceptable values are:
+                - `'pt'`: Return PyTorch `torch.Tensor` objects.
+                - `'np'`: Return Numpy `np.ndarray` objects.
+        """
+        warnings.warn("If you are not getting optimal performence, please try audio sampling_rate as 44100.")
+        warnings.warn(
+            "Pop2PianoFeatureExtractor only takes one raw_audio at a time, if you want to extract features from more than a single audio then you might need to call it multiple times."
```

Comment:
How about we add a check to see if the inputs are batched, and then throw a `ValueError` if they are? E.g.
```python
is_batched = bool(
    isinstance(raw_speech, (list, tuple))
    and (isinstance(raw_speech[0], np.ndarray) or isinstance(raw_speech[0], (tuple, list)))
)

if is_batched:
    raise ValueError("`Pop2PianoFeatureExtractor` got batched audio inputs, but can only take one audio input at a time. To extract features from multiple audio samples, you should call it on the individual audio samples one-by-one.)
```